core-job/src/main/java/org/apache/kylin/job/util/FlatTableSqlQuoteUtils.java

```diff
@@ -25,66 +25,79 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.apache.calcite.sql.SqlDialect;
 import org.apache.commons.lang.StringUtils;
 import org.apache.kylin.common.KylinConfig;
-import org.apache.kylin.common.SourceDialect;
 import org.apache.kylin.metadata.model.ColumnDesc;
 import org.apache.kylin.metadata.model.IJoinedFlatTableDesc;
 import org.apache.kylin.metadata.model.TableDesc;
+import org.apache.kylin.metadata.model.TableRef;
 import org.apache.kylin.metadata.model.TblColRef;
 
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
 
+import static org.apache.calcite.sql.SqlDialect.EMPTY_CONTEXT;
+
 public class FlatTableSqlQuoteUtils {
+
     private static KylinConfig kylinConfig = KylinConfig.getInstanceFromEnv();
+    private static final Map<String, SqlDialect> sqlDialectMap = Maps.newHashMap();
+    private static SqlDialect defaultDialect = null;
```

Comment:
1. `defaultDialect` -> `DEFAULT_DIALECT` to keep all dialects are capitalized
2. Seems to be able to directly declare `DEFAULT_DIALECT = sqlDialectMap.getOrDefault(kylinConfig.getFactTableDialect(), NON_QUOTE_DIALECT)` and remote setQuote function